src/lib/InputComponent.tsx

```diff
@@ -0,0 +1,280 @@
+          } else {
+            this.validationErrors.push(validationResult);
+            this.valid = false;
+          }
+        });
+      } else {
+        let validationResult = this.props.validationFunction(value, this);
+        if (validationResult === true) {
+          this.valid = true;
+        } else {
+          this.validationErrors.push(validationResult);
+          this.valid = false;
+        }
+      }
+    } else if (this.props.keyboardType) {
+      switch (this.props.keyboardType) {
+        case "email-address":
+          validationResult = validateEmail(value);
+          break;
+      }
+      if (validationResult === true) {
+        this.valid = true;
+      } else {
+        this.validationErrors.push(validationResult);
+        this.valid = false;
+      }
+    }
+    this.props.onValidation?.(this.valid, this.validationErrors);
+
+    return this.valid;
+  };
+
+  handleLayoutChange = (e: LayoutChangeEvent) => {
+    if (Platform.OS === "ios") {
+      let { x, y, width, height } = { ...e.nativeEvent.layout };
+
+      this.setState(e.nativeEvent.layout);
+    }
+  };
+
+  handleLabelLayoutChange = e => {
+    if (Platform.OS === "ios") {
+      let { x, y, width, height } = { ...e.nativeEvent.layout };
+
+      this.setState({ labelWidth: width });
+    }
+  };
+
+  handleChange = (event: NativeSyntheticEvent<TextInputChangeEventData>) => {
+    let value = event.nativeEvent.text;
+
+    if (value === this.state.displayValue) {
+      // Skip changeEvent if value is unchanged
+      return;
+    }
+
+    let displayValue: string = this.state.displayValue;
+
+    if (this.props.isDecimal) {
+      const localedValues = this.handleLocale(value);
+      value = localedValues.converted;
+      displayValue = localedValues.display;
+    }
+
+    this.validate(value);
+
+    this.setState({
+      value,
+      displayValue,
+      inputHeight: Math.max(
+        this.state.minFieldHeight,
+        event.nativeEvent.contentSize && this.props.multiline
+          ? event.nativeEvent.contentSize.height
+          : 0
+      )
+    });
+
+    this.props.onChange?.(value, this.valid);
+    this.props.onValueChange?.(value, this.valid);
+  };
+
+  private handleLocale = (value: string) => {
+    const { decimalSeparator, groupingSeparator } = this.props.numberSeparators;
+    const { decimalSeparator: defaultDecimal } = DefaultNumberFormats;
+
+    let converted = value;
+
+    if (decimalSeparator !== defaultDecimal) {
+      // Here we swap , and . to form a correct decimal
+      // e.g. for a european number: 120.000,99 would become 120,000.99
+      converted = value
+        .replace(groupingSeparator, "___")
+        .replace(decimalSeparator, groupingSeparator)
+        .replace("___", decimalSeparator);
+    }
+
+    return {
+      display: value,
+      converted: converted.replace(DefaultNumberFormats.groupingSeparator, "") // commas to form a valid parsable number
```

Comment:
A bit nit-picky, but the grouping separator can also be "." or " ". If you just remove the word "commas" this comment would make more general sense.

Reply:
Fixed in https://github.com/axsy-dev/react-native-form-generator/pull/14/commits/44043b1c6494b356f8adae3775a8f67f277c2a5e